low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server_manifest.py

```diff
@@ -0,0 +1,345 @@
+                os.remove(LSHW_FILE)
+            if os.path.exists(MANIFEST_OUTPUT_FILE):
+                os.remove(MANIFEST_OUTPUT_FILE)
+        except OSError as ex:
+            msg = f"Failed in manifest tmp files cleanup. Error:{str(ex)}"
+            logger.warn(self.log.svc_log(msg))
+        return data
+
+    def get_cortx_service_info(self):
+        """Get cortx service info in required format."""
+        cortx_services = self.service.get_cortx_service_list()
+        cortx_service_info = self.get_service_into(cortx_services)
+        return cortx_service_info
+
+    def get_external_service_info(self):
+        """Get external service info in required format."""
+        external_services = self.service.get_external_service_list()
+        external_service_info = self.get_service_into(external_services)
+        return external_service_info
+
+    def get_service_into(self, services):
+        services_info = []
+        for service in services:
+            response = self.service.get_systemd_service_info(self.log, service)
+            if response is not None:
+                uid, _, health_description, _, specifics = response
+                service_info = {
+                    "uid": uid,
+                    "type": "NA",
+                    "description": health_description,
+                    "product": specifics[0].pop("service_name"),
+                    "manufacturer": "NA",
+                    "serial_number": "NA",
+                    "part_number": "NA",
+                    "version": specifics[0].pop("version"),
+                    "last_updated": int(time.time()),
+                    "specifics": specifics
+                }
+                services_info.append(service_info)
+        return services_info
+
+    def get_bmc_version_info(self):
+        bmc_data = []
+        specifics = {}
+        cmd = "bmc info"
+        out, _, retcode = self.platform._ipmi._run_ipmitool_subcommand(cmd)
+        if retcode == 0:
+            out_lst = out.split("\n")
+            for line in out_lst:
+                data = line.split(':')
+                if len(data)>1 and data[1].strip() != "":
+                    key = data[0].strip().lower().replace(" ","_")
+                    value = data[1].strip()
+                    specifics.update({key: value})
+            bmc = {
+                "uid": 'bmc',
+                "type": "NA",
+                "description": "BMC and IPMI version information",
+                "product": specifics.get("product_name", "NA"),
+                "manufacturer": specifics.get("manufacturer_name", "NA"),
+                "serial_number": "NA",
+                "part_number": "NA",
+                "version": specifics.get("firmware_revision", "NA"),
+                "last_updated": int(time.time()),
+                "specifics": [specifics]
+            }
+            bmc_data.append(bmc)
+        return bmc_data
+    
+    def get_os_server_info(self):
```

Comment:
Suggestion: 
You can create an OS class in `low-level/framework/platforms/server/software.py` and move `get_os_server_info` function over there.

Reply:
Done

Reply:
@satish-seagate, I can still see the function in `server_manifest.py`
So, I am assuming you didn't follow the suggestion?

Reply:
@jjxsg I have moved this function code in https://github.com/satish-seagate/cortx-monitor/blob/8448892590a7ee23a7bf54123a8cf8ce59b9b47f/low-level/framework/platforms/server/platform.py#L33 location and accessing those functions here if you see the function this logic.

Reply:
Okay. Thanks for clarification.